pkg/cmd/cli/cmd/volume.go

```diff
@@ -0,0 +1,588 @@
+	cmd.Flags().StringVar(&opts.ResourceVersion, "resource-version", "", "If non-empty, the volume update will only succeed if this is the current resource-version for the object. Only valid when specifying a single resource")
+	cmd.Flags().StringVarP(&opts.Output, "output", "o", "", "Display the changed objects instead of updating them. One of: json|yaml")
+	cmd.Flags().String("output-version", "", "Output the changed objects with the given version (default api-version).")
+
+	cmd.Flags().StringVarP(&addOpts.Type, "type", "t", "emptyDir", "Type of the volume source for add operation. Supported options: emptyDir, hostPath, secret, persistentVolumeClaim")
+	cmd.Flags().StringVarP(&addOpts.MountPath, "mount-path", "m", "", "Mount path inside the container. Optional param for --add or --remove op")
+	cmd.Flags().StringVar(&addOpts.Path, "path", "", "Host path. Must be provided for hostPath volume type")
+	cmd.Flags().StringVar(&addOpts.SecretName, "secret-name", "", "Name of the persisted secret. Must be provided for secret volume type")
+	cmd.Flags().StringVar(&addOpts.ClaimName, "claim-name", "", "Persistent volume claim name. Must be provided for persistentVolumeClaim volume type")
+	cmd.Flags().StringVar(&addOpts.Source, "source", "", "Details of volume source as json string. This can be used if the required volume type is not supported by --type option. (e.g.: {'gitRepo': {'repository': <git-url>, 'revision': <commit-hash>}})")
+
+	return cmd
+}
+
+func (v *VolumeOptions) Validate(args []string) error {
+	errList := []error{}
+	if len(v.Selector) > 0 {
+		if _, err := labels.Parse(v.Selector); err != nil {
+			errList = append(errList, errors.New("--selector=<selector> must be a valid label selector"))
+		}
+		if v.All {
+			errList = append(errList, errors.New("either specify --selector or --all but not both"))
+		}
+	}
+	if len(args) < 1 {
+		errList = append(errList, errors.New("one ore more resources must be specified as <resource> <name> or <resource>/<name>"))
+	}
+
+	numOps := 0
+	if v.Add {
+		numOps++
+	}
+	if v.Remove {
+		numOps++
+	}
+	if v.List {
+		numOps++
+	}
+	if numOps == 0 {
+		errList = append(errList, errors.New("must provide a volume operation. Support operations: --add, --remove and --list"))
+	} else if numOps > 1 {
+		errList = append(errList, errors.New("must provide only one volume operation at a time"))
+	}
+
+	if v.List && len(v.Output) > 0 {
+		errList = append(errList, errors.New("--list and --output may not be specified together"))
+	}
+	err := v.AddOpts.Validate(v.Add)
+	if err != nil {
+		errList = append(errList, err)
+	}
+	// Removing all volumes for the resource type needs confirmation
+	if v.Remove && len(v.Name) == 0 && !v.Confirm {
+		errList = append(errList, errors.New("must provide --confirm for removing more than one volume"))
+	}
+	return kerrors.NewAggregate(errList)
+}
+
+func (a *AddVolumeOptions) Validate(isAddOp bool) error {
```

Comment:
This method is pretty big, has a lot of nested conditionals, and very little whitespace. Could you at least put some whitespace in between logical units, if nothing else?